sql/core/src/test/scala/org/apache/spark/sql/FileBasedDataSourceSuite.scala

```diff
@@ -72,6 +72,29 @@ class FileBasedDataSourceSuite extends QueryTest with SharedSQLContext {
     }
   }
 
+  // Text and Parquet format does not allow wrting data frame with empty schema.
+  Seq("parquet", "text").foreach { format =>
+    test(s"SPARK-23372 writing empty dataframe should produce AnalysisException - $format") {
+      withTempPath { outputPath =>
+        intercept[AnalysisException] {
+          spark.emptyDataFrame.write.format(format).save(outputPath.toString)
+        }
```

Comment:
Can we check the error message to make it sure?

Reply:
Sure @dongjoon-hyun